### ISSUE TITLE
Lots of slow query warnings for old task deletion

### ISSUE THREAD

**Issue by at055612** [gchq/stroom#3027]
SQL is of the form
delete
from processor_task
where status = 10 AND (create_time_ms is null OR create_time_ms < X)

The method running this sql is `stroom.processor.impl.db.ProcessorTaskDeleteExecutorImpl#deleteOldTasks`

Suspect we need to replace the index `processor_task_status_idx` with the following index

```sql
create index xxx_processor_task_status_create_time_idx
on processor_task (status, create_time_ms); 
```

Worth creating this on ref first to see if it has any positive/negative impact, e.g. gap locks.
This index should also help `stroom.processor.impl.db.ProcessorTaskDaoImpl#retainOwnedTasks` which is running every min or so.

**Comment by at055612** [comment]
In dev testing mysql did not always use this new idx, instead preferring to use the idx on status instead.

**Comment by stroomdev42** [comment]
Also saw the following very similar issues in production system MYSQL slowlog.
delete from ‘processor_task’ where (‘processor_task’.’status’ in (N, N) and (‘processor_task’.’create_time_ms’ is null or ‘processor_task’ . ’create_time_ms’ < 1659950438224 );

**Comment by at055612** [comment]
Did it definitely have `in (N, N) `  or is that a typo?

**Comment by stroomdev42** [comment]
There were several cases with different numbers. I ran mysqldumpslow which grouped them all together, as above (N, N). If you want specifics, I will collect.

**Comment by at055612** [comment]
Don't worry.  The index I have already created should also fit for this case.

**Comment by at055612** [comment]
Fixed in >7.0.1
